File: Python3/3238_find_the_number_of_winning_players.py
# https://leetcode.cn/problems/find-the-number-of-winning-players/

class Solution:
    def winningPlayerCount(self, n, pick):
        """
        Inputs:
            n: int
            pick: list[list[int]]
        Outputs:
            res: int
        """
        res = 0
        win = [0 for _ in range(n)]
        players = [defaultdict(int) for _ in range(n)]
        for a, b in pick:
            if win[a]:
                continue
            players[a][b] += 1
            if players[a][b] > a:
                res += 1
                win[a] = 1
        return res

File: Cpp/3238_find_the_number_of_winning_players.cpp
// https://leetcode.cn/problems/find-the-number-of-winning-players/

class Solution {
public:
    int winningPlayerCount(int n, vector<vector<int>>& pick) {
        int res = 0;
        vector<bool> win(n, false);
        vector<unordered_map<int,int>> players(n);
        for (auto &p: pick) {
            int a = p[0], b = p[1];
            if (win[a]) continue;
            players[a][b] += 1;
            if (players[a][b] > a) {
                res += 1;
                win[a] = true;
            }
        }
        return res;
    }
};
